clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fmap.py

```diff
@@ -0,0 +1,338 @@
+            ("036_S_6088", "bl"),
+            ("036_S_6134", "bl"),
+            ("016_S_6802", "bl"),
+            ("016_S_6816", "bl"),
+            ("126_S_4891", "m84"),
+            # Multiple images
+            ("029_S_2395", "m72"),
+        ]
+
+    # Removing known exceptions from images to convert
+    if not fmap_df.empty:
+        error_ind = fmap_df.index[
+            fmap_df.apply(
+                lambda x: ((x.Subject_ID, x.VISCODE) in conversion_errors), axis=1
+            )
+        ]
+        fmap_df.drop(error_ind, inplace=True)
+
+    # Checking for images paths in filesystem
+    images = find_image_path(fmap_df, source_dir, "FMAP", "S", "Series_ID")
+    images.to_csv(path.join(conversion_dir, "fmap_paths.tsv"), sep="\t", index=False)
+
+    return images
+
+def fmap_image(subject_id, timepoint, visit_str, visit_mri_list, mri_qc_subj):
+    """
+    One image among those in the input list is chosen according to QC
+    and then corresponding metadata is extracted to a dictionary.
+
+    Args:
+        subject_id: Subject identifier
+        timepoint: Visit code
+        visit_str: Visit name
+        visit_mri_list: List of images metadata
+        mri_qc_subj: Dataframe containing list of QC of scans for the subject
+
+    Returns: dictionary - contains image metadata
+    """
+    from clinica.iotools.converters.adni_to_bids.adni_utils import (
+        replace_sequence_chars,
+        select_image_qc,
+    )
+
+    mri_qc_subj.columns = [x.lower() for x in mri_qc_subj.columns]
+    sel_image = select_image_qc(list(visit_mri_list.IMAGEUID), mri_qc_subj)
+
+    if not sel_image:
+        return None
+
+    sel_scan = visit_mri_list[visit_mri_list.IMAGEUID == sel_image].iloc[0]
+
+    image_dict = {
+        "Subject_ID": subject_id,
+        "VISCODE": timepoint,
+        "Visit": visit_str,
+        "Sequence": replace_sequence_chars(sel_scan.SEQUENCE),
+        "Scan_Date": sel_scan["SCANDATE"],
+        "Study_ID": str(int(sel_scan.STUDYID)),
+        "Series_ID": str(int(sel_scan.SERIESID)),
+        "Image_ID": str(int(sel_scan.IMAGEUID)),
+        "Field_Strength": sel_scan.MAGSTRENGTH,
+    }
+
+    return image_dict
+
+def rename_fmaps(destination_dir):
+    import os
+    import shutil
+    import json
+
+    dir_name = destination_dir
+    for root, dirs, _ in os.walk(destination_dir):
+        for dir_name in dirs:
+            if dir_name.startswith('tmp_dcm'):
+                dir_path = os.path.join(root, dir_name)
+                shutil.rmtree(dir_path)
+
+    for root, dirs, _ in os.walk(destination_dir):
+        for dir_name in dirs:
+            if dir_name.startswith('sub-ADNI'):
+                for subroot, subdir, filenames in os.walk(os.path.join(root, dir_name)):
+                    for filename in filenames:
+                        fmap_filename = os.path.join(subroot, filename)
+                        if filename[-11:] == "fmap.nii.gz" and os.path.exists(fmap_filename):
```

Comment:
I'm very concerned about the maintainability of this giant for loop...
It's a bit hard for me to tell, but I believe we can come up with simpler ways to rename these files.
It would be helpful to have examples of the tasks the function is supposed to perform.